src/grid/hybrid/grid_collocate_hybrid.c

```diff
@@ -0,0 +1,505 @@
+/*----------------------------------------------------------------------------*/
+/*  CP2K: A general program to perform molecular dynamics simulations         */
+/*  Copyright 2000-2020 CP2K developers group <https://cp2k.org>              */
+/*                                                                            */
+/*  SPDX-License-Identifier: GPL-2.0-or-later                                 */
+/*----------------------------------------------------------------------------*/
+
+#ifdef __COLLOCATE_GPU
+
+#include <assert.h>
+#include <cuda.h>
```

Comment:
The Cuda headers can not be included into C file, because Cuda assumes C++. Hence, you can only use them in a `.cu` file.